confluent_kafka_helpers/consumer.py

```diff
@@ -123,16 +134,47 @@ def _message_generator(self):
                 continue
 
             statsd.increment(f'{base_metric}.consumer.message.count.total')
-            message = Message(message)
 
-            with tracer.extract_headers_and_start_span(
-                operation_name='kafka.consumer.consume', headers=message._meta.headers
+            value, resource_name = message.value(), message.topic()
+            message_class = value.get("class") if isinstance(value, dict) else None
+            resource_name = f"{message.topic()}:{message_class}" if message_class else resource_name
+
+            context = tracer.extract_headers(headers=decode_kafka_headers(message.headers()))
```

Comment:
Extract the tracing headers injected as Kafka headers and use them in the context if this span. This is what makes the trace "distributed".